presto-main/src/main/java/com/facebook/presto/operator/aggregation/MergeStatisticalDigestFunction.java

```diff
@@ -0,0 +1,146 @@
+
+import com.facebook.presto.bytecode.DynamicClassLoader;
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionManager;
+import com.facebook.presto.metadata.SqlAggregationFunction;
+import com.facebook.presto.operator.aggregation.state.StatisticalDigestState;
+import com.facebook.presto.operator.aggregation.state.StatisticalDigestStateFactory;
+import com.facebook.presto.operator.aggregation.state.StatisticalDigestStateSerializer;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.spi.type.TypeSignatureParameter;
+import com.facebook.presto.type.StatisticalDigest;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.AccumulatorStateDescriptor;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.spi.function.Signature.comparableTypeParameter;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.util.Reflection.methodHandle;
+
+public abstract class MergeStatisticalDigestFunction
+        extends SqlAggregationFunction
+{
+    public final String name;
+    public final String type;
+    public final StatisticalDigestStateFactory factory;
+
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(
+            MergeStatisticalDigestFunction.class,
+            "combine",
+            StatisticalDigestState.class,
+            StatisticalDigestState.class);
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(
+            MergeStatisticalDigestFunction.class,
+            "output",
+            StatisticalDigestStateSerializer.class,
+            StatisticalDigestState.class,
+            BlockBuilder.class);
+
+    MergeStatisticalDigestFunction(String name, String type, StatisticalDigestStateFactory factory)
+    {
+        super(name,
+                ImmutableList.of(comparableTypeParameter("T")),
+                ImmutableList.of(),
+                parseTypeSignature(type + "(T)"),
+                ImmutableList.of(parseTypeSignature(type + "(T)")));
+        this.name = name;
+        this.type = type;
+        this.factory = factory;
+    }
+
+    @Override
+    public String getDescription()
```

Comment:
Does it make more sense to have the two different implementations to have different description? I think you can remove this function here.

Reply:
Replacing with the following:
```
return format("Merges the input %1$ss into a single %1$s", type);
```

Reply:
I would go with:

`return format("Merges the input %s into a single %s", type, type);`